### PR TITLE
Add epoll_wait_or_signal to std.os

### PR DESCRIPTION
**Pull request by okx-code**
It is not possible to handle signals in `std.os.epoll_wait` - they are simply retried - so one must resort to `std.linux.epoll_pwait`, where the interface is not as nice to use. This PR adds a method that allows the user to handle signals more elegantly.

### CODE REVIEW

File: lib/std/os.zig [hunk 1]
```diff
@@ -3815,20 +3815,31 @@ pub fn epoll_ctl(epfd: i32, op: u32, fd: i32, event: ?*linux.epoll_event) EpollC
...
+        } else |_| {}
+    }
+}
+
+/// Waits for an I/O event on an epoll file descriptor.
+/// Returns an error if interrupted by a signal.
+/// Returns the number of file descriptors ready for the requested I/O,
+/// or zero if no file descriptor became ready during the requested timeout milliseconds.
+pub fn epoll_pwait(epfd: i32, events: []linux.epoll_event, timeout: i32, sigset: ?*const linux.sigset_t) !usize {
+    // TODO get rid of the @intCast
```
Comment (matu3ba) [inline lib/std/os.zig:3834]: Please fix the TODO.

File: lib/std/os.zig [hunk 2]
```diff
@@ -3815,20 +3815,31 @@ pub fn epoll_ctl(epfd: i32, op: u32, fd: i32, event: ?*linux.epoll_event) EpollC
...
-            else => unreachable,
-        }
+        if (epoll_pwait(epfd, events, timeout, null)) |value| {
+            return value;
+        } else |_| {}
+    }
+}
+
+/// Waits for an I/O event on an epoll file descriptor.
+/// Returns an error if interrupted by a signal.
```
Comment (matu3ba) [inline lib/std/os.zig:3830]: It's only one error, so be explicit which one.

File: lib/std/os.zig [hunk 3]
```diff
@@ -3815,20 +3815,31 @@ pub fn epoll_ctl(epfd: i32, op: u32, fd: i32, event: ?*linux.epoll_event) EpollC
...
+/// Returns the number of file descriptors ready for the requested I/O,
+/// or zero if no file descriptor became ready during the requested timeout milliseconds.
```
Comment (matu3ba) [inline lib/std/os.zig:3832]: s/timeout millisecond./timeout in milliseconds.

**Review by matu3ba — COMMENTED** [review]
Some suggestions and this is missing a test for epoll_pwait returning the error.

Probably, a core contributors should come up with a naming schema or organization for functions with non erroneous and reachable interrupt errors.

File: lib/std/os.zig (hunk 1 above)
Comment (okx-code) [reply lib/std/os.zig:3834]: I'm honestly not sure why this was added in the first place, there doesn't seem to be a better method.

**Comment by andrewrk** [comment]
Please file an issue instead. I think this should be solved a different way, and this pull request was never passing CI checks, nor was it merge ready in its current state.

**Pull request closed without merging**